stl/inc/utility

```diff
@@ -644,6 +645,77 @@ template <size_t _Idx>
 inline constexpr in_place_index_t<_Idx> in_place_index{};
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+
+template <typename _Ty>
+inline constexpr bool _Is_integral_not_bool_char = is_integral_v<_Ty>
+    && !is_same_v<_Ty, bool> && !is_same_v<_Ty, char>
+    && !is_same_v<_Ty, char8_t> && !is_same_v<_Ty, char16_t>
+    && !is_same_v<_Ty, char32_t> && !is_same_v<_Ty, wchar_t>;
+
+// SAFE INTEGRAL COMPARISONS
+template <typename _Ty1, typename _Ty2>
+constexpr bool in_range(const _Ty2 _Value) noexcept {
+    static_assert(_Is_integral_not_bool_char<_Ty1> && _Is_integral_not_bool_char<_Ty2>);
+    if constexpr (is_signed_v<_Ty1> && is_unsigned_v<_Ty2>) {
+        return _Value <= (numeric_limits<_Ty1>::max)();

```

Comment:
This is a signed / unsigned mismatch.
You can cast numeric_limit<_Ty1>::max() to make_unsiged<_Ty1> to silence compiler warnings.

Note that it's possible to base the implementation on the other function templates.
`return cmp_greater_equal(_Value, (numeric_limits<_Ty1>::min()) && (cmp_less_equal(_Value, numeric_limits<_Ty1>::max());`

Reply:
fix in new commit